azurerm/data_source_network_interface.go

```diff
@@ -0,0 +1,273 @@
+package azurerm
+
+import (
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/services/network/mgmt/2017-09-01/network"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmNetworkInterface() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmNetworkInterfaceRead,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+
+			"location": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
+			"resource_group_name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
can we switch over to using the helper for this:

```
"resource_group_name": resourceGroupNameForDataSourceSchema(),
```

Reply:
I was not sure it was wise to use it for data source but this makes sense. done.